compositions/NFTPage/NFTInfoSidebar.tsx

```diff
@@ -1,25 +1,50 @@
-import { Heading, Stack, Flex, Paragraph, Box, BoxProps } from '@zoralabs/zord'
+import { Heading, Stack, Flex, Paragraph, Box, BoxProps, Button } from '@zoralabs/zord'
 import { CollectionThumbnail } from '@media/CollectionThumbnail'
-import { useNFTProvider, useTitleWithFallback } from '@shared'
+import { useIsOwner, useNFTProvider, useTitleWithFallback } from '@shared'
 import { Link } from 'components'
 import { clickAnimation } from 'styles/styles.css'
-import { nftInfoSidebar, nftInfoSidebarWrapper } from './NFTPage.css'
+import { nftInfoSidebarWrapper, nftNextButton, nftInfoSidebar } from './NFTPage.css'
 import { MarketUi } from './MarketUi'
 
 import { lightFont } from '@shared'
+import { useRouter } from 'next/router'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { useNounishAuctionProvider } from '@noun-auction'
 
 export interface NFTInfoSidebar extends BoxProps {}
 
 export function NFTInfoSidebar({ ...props }: NFTInfoSidebar) {
+  const router = useRouter()
   const { initialData: nft, tokenId, contractAddress } = useNFTProvider()
+  const { activeAuctionId } = useNounishAuctionProvider()
 
-  if (!nft || !tokenId || !contractAddress) return null
+  const hasNextNft = useMemo(() => {
+    return tokenId && activeAuctionId
+      ? !!(parseInt(tokenId) < parseInt(activeAuctionId))
+      : false
+  }, [activeAuctionId, tokenId])
+
+  const hasPrevoiusNft = useMemo(() => {
+    return !!tokenId && parseInt(tokenId) >= 0
+  }, [tokenId])
+
+  const handleNext = useCallback(() => {
+    if (hasNextNft) {
+      tokenId && router.push(`/collections/${contractAddress}/${parseInt(tokenId) + 1}`)
```

Comment:
Are you verifying that there is a next tokenId in the collection? It looks like hasNextNFT is checking against the auctionId and not the collection token count.

Reply:
`hasNextNft` does it